samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/assessment/AssessmentService.java

```diff
@@ -1118,6 +1118,191 @@ public SecurityAdvice isAllowed(String arg0, String arg1,
 			return text;
 		}
 
+
+	/**
+	 * 
+	 * @param assessment
+	 * @param bundle
+	 * @return
+	 */
+	public String exportAssessmentToMarkupText(AssessmentFacade assessment, Map<String,String> bundle) {
+		StringBuffer markupText = new StringBuffer(); 
+		int nQuestion = 1;
+		
+		Iterator<SectionFacade> sectionIter = assessment.getSectionArray().iterator();
+		while (sectionIter.hasNext()) {
+			SectionFacade section = sectionIter.next();
+			List<ItemDataIfc> items;
+			boolean hasRandomPartScore = false;
+			Double score = null;
+			boolean hasRandomPartDiscount = false;
+			Double discount = null;
+			
+			if (section.getSectionMetaDataByLabel(SectionDataIfc.AUTHOR_TYPE).equals(SectionDataIfc.QUESTIONS_AUTHORED_ONE_BY_ONE.toString()))
+  			{
+				items = section.getItemArray();
+  			}
+			else 
+			{
+				String requestedScore = (section.getSectionMetaDataByLabel(SectionDataIfc.POINT_VALUE_FOR_QUESTION) != null) ? 
+						                 section.getSectionMetaDataByLabel(SectionDataIfc.POINT_VALUE_FOR_QUESTION)	: "";
+						                 
+				if (requestedScore != null && !requestedScore.equals("")) {
+					hasRandomPartScore = true;
+					score = new Double(requestedScore);
```

Comment:
This constructor has the possibility of throwing a `NumberFormatException`. It may be highly unlikely, but you should still catch it.